diff --git a/engine/src/main/java/org/hibernate/validator/internal/metadata/core/BuiltinConstraint.java b/engine/src/main/java/org/hibernate/validator/internal/metadata/core/BuiltinConstraint.java
index be445497c..5d6861726 100644
--- a/engine/src/main/java/org/hibernate/validator/internal/metadata/core/BuiltinConstraint.java
+++ b/engine/src/main/java/org/hibernate/validator/internal/metadata/core/BuiltinConstraint.java
@@ -54,6 +54,7 @@ enum BuiltinConstraint {
 	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_CODE_POINT_LENGTH( "org.hibernate.validator.constraints.CodePointLength" ),
 	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_CURRENCY( "org.hibernate.validator.constraints.Currency" ),
 	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_IP_ADDRESS( "org.hibernate.validator.constraints.IpAddress" ),
+	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_ID_CARD( "org.hibernate.validator.constraints.IdCard" ),
 	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_ISBN( "org.hibernate.validator.constraints.ISBN" ),
 	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_LENGTH( "org.hibernate.validator.constraints.Length" ),
 	ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_LUHN_CHECK( "org.hibernate.validator.constraints.LuhnCheck" ),
diff --git a/engine/src/main/java/org/hibernate/validator/internal/metadata/core/ConstraintHelper.java b/engine/src/main/java/org/hibernate/validator/internal/metadata/core/ConstraintHelper.java
index cb6585b52..1251448b2 100644
--- a/engine/src/main/java/org/hibernate/validator/internal/metadata/core/ConstraintHelper.java
+++ b/engine/src/main/java/org/hibernate/validator/internal/metadata/core/ConstraintHelper.java
@@ -34,6 +34,7 @@
 import static org.hibernate.validator.internal.metadata.core.BuiltinConstraint.ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_CREDIT_CARD_NUMBER;
 import static org.hibernate.validator.internal.metadata.core.BuiltinConstraint.ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_CURRENCY;
 import static org.hibernate.validator.internal.metadata.core.BuiltinConstraint.ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_EAN;
+import static org.hibernate.validator.internal.metadata.core.BuiltinConstraint.ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_ID_CARD;
 import static org.hibernate.validator.internal.metadata.core.BuiltinConstraint.ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_IP_ADDRESS;
 import static org.hibernate.validator.internal.metadata.core.BuiltinConstraint.ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_ISBN;
 import static org.hibernate.validator.internal.metadata.core.BuiltinConstraint.ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_KOR_KORRRN;
@@ -110,6 +111,7 @@
 import org.hibernate.validator.constraints.Currency;
 import org.hibernate.validator.constraints.EAN;
 import org.hibernate.validator.constraints.ISBN;
+import org.hibernate.validator.constraints.IdCard;
 import org.hibernate.validator.constraints.IpAddress;
 import org.hibernate.validator.constraints.Length;
 import org.hibernate.validator.constraints.LuhnCheck;
@@ -333,6 +335,7 @@
 import org.hibernate.validator.internal.constraintvalidators.hv.CodePointLengthValidator;
 import org.hibernate.validator.internal.constraintvalidators.hv.EANValidator;
 import org.hibernate.validator.internal.constraintvalidators.hv.ISBNValidator;
+import org.hibernate.validator.internal.constraintvalidators.hv.IdCardValidator;
 import org.hibernate.validator.internal.constraintvalidators.hv.IpAddressValidator;
 import org.hibernate.validator.internal.constraintvalidators.hv.LengthValidator;
 import org.hibernate.validator.internal.constraintvalidators.hv.LuhnCheckValidator;
@@ -777,6 +780,9 @@ protected Map<Class<? extends Annotation>, List<? extends ConstraintValidatorDes
 		if ( enabledBuiltinConstraints.contains( ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_IP_ADDRESS ) ) {
 			putBuiltinConstraint( tmpConstraints, IpAddress.class, IpAddressValidator.class );
 		}
+		if ( enabledBuiltinConstraints.contains( ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_ID_CARD ) ) {
+			putBuiltinConstraint( tmpConstraints, IdCard.class, IdCardValidator.class );
+		}
 		if ( enabledBuiltinConstraints.contains( ORG_HIBERNATE_VALIDATOR_CONSTRAINTS_ISBN ) ) {
 			putBuiltinConstraint( tmpConstraints, ISBN.class, ISBNValidator.class );
 		}
diff --git a/engine/src/main/resources/org/hibernate/validator/ValidationMessages.properties b/engine/src/main/resources/org/hibernate/validator/ValidationMessages.properties
index 98aef1f1d..49005b57d 100644
--- a/engine/src/main/resources/org/hibernate/validator/ValidationMessages.properties
+++ b/engine/src/main/resources/org/hibernate/validator/ValidationMessages.properties
@@ -25,6 +25,7 @@ org.hibernate.validator.constraints.CreditCardNumber.message        = invalid cr
 org.hibernate.validator.constraints.Currency.message                = invalid currency (must be one of {value})
 org.hibernate.validator.constraints.EAN.message                     = invalid {type} barcode
 org.hibernate.validator.constraints.IpAddress.message               = invalid IP address
+org.hibernate.validator.constraints.IdCard.message                  = invalid Chinese Resident Identity Card number
 org.hibernate.validator.constraints.ISBN.message                    = invalid ISBN
 org.hibernate.validator.constraints.Length.message                  = length must be between {min} and {max}
 org.hibernate.validator.constraints.CodePointLength.message         = length must be between {min} and {max}
diff --git a/engine/src/main/resources/org/hibernate/validator/ValidationMessages_zh.properties b/engine/src/main/resources/org/hibernate/validator/ValidationMessages_zh.properties
index 0b8998145..970f81e54 100644
--- a/engine/src/main/resources/org/hibernate/validator/ValidationMessages_zh.properties
+++ b/engine/src/main/resources/org/hibernate/validator/ValidationMessages_zh.properties
@@ -24,6 +24,7 @@ jakarta.validation.constraints.Size.message            = \u5927\u5c0f\u5fc5\u987
 org.hibernate.validator.constraints.CreditCardNumber.message        = \u65e0\u6548\u4fe1\u7528\u5361\u5361\u53f7
 org.hibernate.validator.constraints.Currency.message                = \u65e0\u6548\u8d27\u5e01\uff08\u5fc5\u987b\u4e3a {value} \u4e4b\u4e00\uff09
 org.hibernate.validator.constraints.EAN.message                     = \u65e0\u6548 {type} \u6761\u5f62\u7801
+org.hibernate.validator.constraints.IdCard.message                  = \u65e0\u6548\u7684\u4e2d\u56fd\u5c45\u6c11\u8eab\u4efd\u8bc1\u53f7\u7801
 org.hibernate.validator.constraints.ISBN.message                    = \u65e0\u6548 ISBN
 org.hibernate.validator.constraints.Length.message                  = \u957f\u5ea6\u5fc5\u987b\u4ecb\u4e8e {min} \u4e0e {max} \u4e4b\u95f4
 org.hibernate.validator.constraints.CodePointLength.message         = \u957f\u5ea6\u5fc5\u987b\u4ecb\u4e8e {min} \u4e0e {max} \u4e4b\u95f4
