diff --git a/engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanHelper.java b/engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanHelper.java
index 640b1394d..3f14c7c5e 100644
--- a/engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanHelper.java
+++ b/engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanHelper.java
@@ -10,6 +10,7 @@
 import java.lang.reflect.Executable;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
+import java.lang.reflect.RecordComponent;
 import java.util.Optional;
 
 import org.hibernate.validator.internal.properties.Constrainable;
@@ -21,15 +22,10 @@
 import org.hibernate.validator.spi.nodenameprovider.JavaBeanProperty;
 import org.hibernate.validator.spi.nodenameprovider.PropertyNodeNameProvider;
 import org.hibernate.validator.spi.nodenameprovider.PropertyNodeNameProviderContext;
+import org.hibernate.validator.spi.nodenameprovider.RecordComponentProperty;
 import org.hibernate.validator.spi.properties.ConstrainableExecutable;
 import org.hibernate.validator.spi.properties.GetterPropertySelectionStrategy;
 
-/**
- * Helper class that gives ability to find {@link Constrainable} versions
- * of JavaBean's fields, getters, constructors and methods.
- *
- * @author Marko Bekhta
- */
 public class JavaBeanHelper implements PropertyNodeNameProviderContext {
 
 	private final GetterPropertySelectionStrategy getterPropertySelectionStrategy;
@@ -78,8 +74,7 @@ private Optional<JavaBeanGetter> findGetter(Class<?> declaringClass, String prop
 			return Optional.empty();
 		}
 		else {
-			return Optional.of( new JavaBeanGetter( declaringClass, getter, property, propertyNodeNameProvider.getName(
-					new JavaBeanPropertyImpl( declaringClass, property, getter.getName() ), this ) ) );
+			return Optional.of( new JavaBeanGetter( declaringClass, getter, property, resolvePropertyNodeName( declaringClass, property, getter.getName() ) ) );
 		}
 	}
 
@@ -120,15 +115,39 @@ public JavaBeanMethod executable(Class<?> declaringClass, Method method) {
 
 		Optional<String> correspondingProperty = getterPropertySelectionStrategy.getProperty( executable );
 		if ( correspondingProperty.isPresent() ) {
-			return new JavaBeanGetter( declaringClass, method, correspondingProperty.get(), propertyNodeNameProvider.getName(
-					new JavaBeanPropertyImpl( declaringClass, correspondingProperty.get(), method.getName() ), this ) );
+			return new JavaBeanGetter( declaringClass, method, correspondingProperty.get(), resolvePropertyNodeName( declaringClass, correspondingProperty.get(), method.getName() ) );
 		}
 
 		return new JavaBeanMethod( method );
 	}
 
 	public JavaBeanField field(Field field) {
-		return new JavaBeanField( field, propertyNodeNameProvider.getName( new JavaBeanPropertyImpl( field.getDeclaringClass(), field.getName(), field.getName() ), this ) );
+		return new JavaBeanField( field, resolvePropertyNodeName( field.getDeclaringClass(), field.getName(), field.getName() ) );
+	}
+
+	private String resolvePropertyNodeName(Class<?> declaringClass, String propertyName, String memberName) {
+		return propertyNodeNameProvider.getName( createPropertyMetadata( declaringClass, propertyName, memberName ), this );
+	}
+
+	private org.hibernate.validator.spi.nodenameprovider.Property createPropertyMetadata(Class<?> declaringClass, String propertyName, String memberName) {
+		if ( isRecordComponent( declaringClass, propertyName ) ) {
+			return new RecordComponentPropertyImpl( declaringClass, propertyName, memberName );
+		}
+		return new JavaBeanPropertyImpl( declaringClass, propertyName, memberName );
+	}
+
+	private static boolean isRecordComponent(Class<?> declaringClass, String propertyName) {
+		if ( !declaringClass.isRecord() ) {
+			return false;
+		}
+
+		for ( RecordComponent recordComponent : declaringClass.getRecordComponents() ) {
+			if ( recordComponent.getName().equals( propertyName ) ) {
+				return true;
+			}
+		}
+
+		return false;
 	}
 
 	private static class JavaBeanConstrainableExecutable implements ConstrainableExecutable {
@@ -181,4 +200,31 @@ public String getName() {
 			return name;
 		}
 	}
+
+	private static class RecordComponentPropertyImpl implements RecordComponentProperty, JavaBeanProperty {
+		private final Class<?> declaringClass;
+		private final String name;
+		private final String memberName;
+
+		private RecordComponentPropertyImpl(Class<?> declaringClass, String name, String memberName) {
+			this.declaringClass = declaringClass;
+			this.name = name;
+			this.memberName = memberName;
+		}
+
+		@Override
+		public Class<?> getDeclaringClass() {
+			return declaringClass;
+		}
+
+		@Override
+		public String getMemberName() {
+			return memberName;
+		}
+
+		@Override
+		public String getName() {
+			return name;
+		}
+	}
 }
diff --git a/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/AnnotationPropertyNodeNameProvider.java b/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/AnnotationPropertyNodeNameProvider.java
index d3d8ee760..b76fb1048 100644
--- a/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/AnnotationPropertyNodeNameProvider.java
+++ b/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/AnnotationPropertyNodeNameProvider.java
@@ -9,6 +9,7 @@
 import java.lang.invoke.MethodHandles;
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.RecordComponent;
 import java.util.Arrays;
 import java.util.Objects;
 import java.util.Optional;
@@ -19,12 +20,8 @@
 import org.hibernate.validator.spi.nodenameprovider.Property;
 import org.hibernate.validator.spi.nodenameprovider.PropertyNodeNameProvider;
 import org.hibernate.validator.spi.nodenameprovider.PropertyNodeNameProviderContext;
+import org.hibernate.validator.spi.nodenameprovider.RecordComponentProperty;
 
-/**
- * An example of how a name can be resolved from an annotation
- *
- * @author Damir Alibegovic
- */
 class AnnotationPropertyNodeNameProvider implements PropertyNodeNameProvider, Serializable {
 	private static final String VALUE = "value";
 	private static final Log LOG = LoggerFactory.make( MethodHandles.lookup() );
@@ -43,6 +40,10 @@ class AnnotationPropertyNodeNameProvider implements PropertyNodeNameProvider, Se
 
 	@Override
 	public String getName(Property property, PropertyNodeNameProviderContext context) {
+		if ( property instanceof RecordComponentProperty recordComponentProperty ) {
+			return getRecordComponentPropertyName( recordComponentProperty );
+		}
+
 		if ( property instanceof JavaBeanProperty javaBeanProperty ) {
 			return getJavaBeanPropertyName( javaBeanProperty );
 		}
@@ -62,6 +63,18 @@ private String getJavaBeanPropertyName(JavaBeanProperty property) {
 		}
 	}
 
+	private String getRecordComponentPropertyName(RecordComponentProperty property) {
+		Optional<RecordComponent> recordComponent = getRecordComponent( property );
+
+		if ( recordComponent.isPresent() && recordComponent.get().isAnnotationPresent( annotationType ) ) {
+			return getAnnotationMemberValue( recordComponent.get(), annotationMemberName )
+					.orElse( getDefaultName( property ) );
+		}
+		else {
+			return getDefaultName( property );
+		}
+	}
+
 	private Optional<Field> getField(JavaBeanProperty property) {
 		return Arrays.stream( property.getDeclaringClass().getFields() )
 				.peek( field -> field.setAccessible( true ) )
@@ -69,8 +82,18 @@ private Optional<Field> getField(JavaBeanProperty property) {
 				.findFirst();
 	}
 
-	private Optional<String> getAnnotationMemberValue(Field field, String annotationMemberName) {
-		Annotation annotation = field.getAnnotation( annotationType );
+	private Optional<RecordComponent> getRecordComponent(RecordComponentProperty property) {
+		if ( !property.getDeclaringClass().isRecord() ) {
+			return Optional.empty();
+		}
+
+		return Arrays.stream( property.getDeclaringClass().getRecordComponents() )
+				.filter( recordComponent -> property.getName().equals( recordComponent.getName() ) )
+				.findFirst();
+	}
+
+	private Optional<String> getAnnotationMemberValue(AnnotatedElementWrapper annotatedElementWrapper, String annotationMemberName) {
+		Annotation annotation = annotatedElementWrapper.getAnnotation( annotationType );
 
 		try {
 			return Optional.of(
@@ -83,7 +106,45 @@ private Optional<String> getAnnotationMemberValue(Field field, String annotation
 		}
 	}
 
+	private Optional<String> getAnnotationMemberValue(Field field, String annotationMemberName) {
+		return getAnnotationMemberValue( new FieldWrapper( field ), annotationMemberName );
+	}
+
+	private Optional<String> getAnnotationMemberValue(RecordComponent recordComponent, String annotationMemberName) {
+		return getAnnotationMemberValue( new RecordComponentWrapper( recordComponent ), annotationMemberName );
+	}
+
 	private String getDefaultName(Property property) {
 		return property.getName();
 	}
+
+	private interface AnnotatedElementWrapper {
+		Annotation getAnnotation(Class<? extends Annotation> annotationType);
+	}
+
+	private static class FieldWrapper implements AnnotatedElementWrapper {
+		private final Field field;
+
+		private FieldWrapper(Field field) {
+			this.field = field;
+		}
+
+		@Override
+		public Annotation getAnnotation(Class<? extends Annotation> annotationType) {
+			return field.getAnnotation( annotationType );
+		}
+	}
+
+	private static class RecordComponentWrapper implements AnnotatedElementWrapper {
+		private final RecordComponent recordComponent;
+
+		private RecordComponentWrapper(RecordComponent recordComponent) {
+			this.recordComponent = recordComponent;
+		}
+
+		@Override
+		public Annotation getAnnotation(Class<? extends Annotation> annotationType) {
+			return recordComponent.getAnnotation( annotationType );
+		}
+	}
 }
diff --git a/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/PropertyNodeNameProviderTest.java b/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/PropertyNodeNameProviderTest.java
index 3478ea022..2e0080cbf 100644
--- a/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/PropertyNodeNameProviderTest.java
+++ b/engine/src/test/java/org/hibernate/validator/test/spi/nodenameprovider/PropertyNodeNameProviderTest.java
@@ -5,6 +5,7 @@
 package org.hibernate.validator.test.spi.nodenameprovider;
 
 import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertThat;
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.pathWith;
 import static org.hibernate.validator.testutil.ConstraintViolationAssert.violationOf;
 import static org.testng.Assert.assertEquals;
 
@@ -22,6 +23,7 @@
 import jakarta.validation.Validator;
 import jakarta.validation.ValidatorFactory;
 import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
 import jakarta.validation.constraints.Size;
 
 import org.hibernate.validator.HibernateValidator;
@@ -33,9 +35,6 @@
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
-/**
- * @author Damir Alibegovic
- */
 @TestForIssue(jiraKey = "HV-823")
 public class PropertyNodeNameProviderTest {
 	private static final String INVALID_BRAND_NAME = "BMW";
@@ -107,6 +106,28 @@ public void propertyCanBeOfTypeMap() {
 
 	}
 
+	@Test
+	public void nameIsResolvedFromRecordComponentsInNestedCascadedValidation() {
+		ValidatorFactory validatorFactory = Validation.byProvider( HibernateValidator.class )
+				.configure()
+				.propertyNodeNameProvider( new AnnotationPropertyNodeNameProvider( RecordPropertyName.class ) )
+				.buildValidatorFactory();
+		Validator recordValidator = validatorFactory.getValidator();
+
+		Set<ConstraintViolation<RecordCar>> violations = recordValidator.validate(
+				new RecordCar( new RecordEngine( new RecordCylinder( "" ) ) )
+		);
+		ConstraintViolation<RecordCar> violation = violations.iterator().next();
+
+		assertEquals( violation.getPropertyPath().toString(), "outer.inner.field" );
+		assertThat( violations ).containsOnlyViolations(
+				violationOf( NotBlank.class ).withPropertyPath( pathWith()
+						.property( "outer" )
+						.property( "inner" )
+						.property( "field" ) )
+		);
+	}
+
 	@Test
 	public void defaultValidatorFactoryUsesDefaultPropertyNodeNameProvider() {
 		ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
@@ -212,6 +233,12 @@ public void constraintsDefinedProgrammatically() {
 		String value();
 	}
 
+	@Retention(RetentionPolicy.RUNTIME)
+	@Target({ ElementType.RECORD_COMPONENT })
+	public @interface RecordPropertyName {
+		String value();
+	}
+
 	private class Car {
 		@PropertyName("components")
 		public final Map<String, @Valid Object> comps = new HashMap<>();
@@ -275,4 +302,13 @@ private static class ProgrammaticCar {
 			this.brand = brand;
 		}
 	}
+
+	private record RecordCar(@Valid @RecordPropertyName("outer") RecordEngine outerRecord) {
+	}
+
+	private record RecordEngine(@Valid @RecordPropertyName("inner") RecordCylinder innerRecord) {
+	}
+
+	private record RecordCylinder(@NotBlank String field) {
+	}
 }
